Naprawione problemy wspomniane w Issues

darkassazi opened the pull request.

darkassazi committed:
Extract all US1 requirements to another class

darkassazi committed:
Create Tests for US1 & Extract user input part into separate function

darkassazi committed:
Update .travis.yml

darkassazi committed:
change getNick to getName to validate that object contain data

darkassazi committed:
Merge branch 'Tylko_Nie_Kanbanosy_Fix' of https://github.com/coders-school/CourseDashboard into Tylko_Nie_Kanbanosy_Fix

ziobron commented on src/User.hpp:
```diff
@@ -18,8 +19,13 @@ class User
 
+    const std::string& getName() const;
+
+    User& operator=(const User& user) = default;
```
Why do we want to copy a user? I wouldn't like to do it :)

darkassazi replied:
User should be copyable because the UserContainer have ownership of the user and only via provided methods Update and delete user object should be changed. 

This mean that retrive user need to return copy of an object rather than orginal object.

darkassazi replied:
corrected

ziobron commented on src/UserContainer.cpp:
```diff
@@ -0,0 +1,58 @@
+    });
+
+    *it = user;
+}
```
what if we would like to update the nick?

darkassazi replied:
Then user will be considered a new user and update will not happen.

darkassazi replied:
To change this i need to implement an unique and unchanging by user field in user model so even after all user data are changed i will be able to locate given user

darkassazi replied:
corrected

ziobron commented on src/UserContainer.hpp:
```diff
@@ -0,0 +1,27 @@
+
+    std::stringstream  showAll();
+
+    void createUser(const User & user);
```
I don't like this name. It says createUser, but it takes already created user as an argument. I suggest to change the name to put/emplace/pushUser (your choice).

darkassazi replied:
it's not my name :)
I'm just cleaning here.

darkassazi replied:
So please create an Issue so I can formally change the name of the method

darkassazi replied:
I suggest "add" because we add user.

ziobron commented on tests/UserContainerTests.cpp:
```diff
@@ -0,0 +1,68 @@
+
+TEST_F(UserContainerTests, canConstructUserContainer)
+{
+    UserContainer uc;
```
test not needed, because other tests always construct UserContainer, so it is covered by them.

darkassazi replied:
Fixed

ziobron commented on tests/UserContainerTests.cpp:
```diff
@@ -0,0 +1,68 @@
+    
+    auto result = userContainer_.retriveUserByNick("SzymonGajewski");
+    ASSERT_TRUE(result.has_value());  
+    ASSERT_TRUE(!result.value().getName().compare("Szymon"));
```
you can implement comparison operator and just compare users.

darkassazi replied:
implemented

darkassazi replied:
Implemented

ziobron commented on tests/UserContainerTests.cpp:
```diff
@@ -0,0 +1,68 @@
+    result.value().setNick("Kamil.Waszkiewicz");
+    result.value().setGroup("weekendowa");
+    result.value().setGitHub("Kamil.Waszkiewicz");
+    result.value().setFirecode("SzymonGajewski");
```
`result->set...` should work I guess

darkassazi replied:
fixed

darkassazi committed:
-Implemented: operator==
-Added tests for User
-Change Name of metod from createUser to add in  UserContainer

darkassazi committed:
user can only be moved, add exceptions, nick can be changed

darkassazi committed:
Add last tests
